dolosse/run_control/cli/kafka_pyinq.py

```diff
@@ -0,0 +1,246 @@
+            else:
+                print('Files check out!')
+                with open(equipment_filename) as ef:
+                    equipment_json = json.dumps(json.loads(ef.read()))
+                with open(metadata_filename) as mf:
+                    metadata_json = json.dumps(json.loads(mf.read()))
+
+                # compose control message
+                control_json = json.dumps({"category": "control", "run": {"action": answers['command'],
+                                                                          "run_number": run_number,
+                                                                          "date": date}})
+                control_msg = Message(topic_control, control_json)
+                control_msg.execute()
+
+                manage_msg = Message(topic_manage, control_json)
+                manage_msg.execute()
+
+                manage_msg.json = equipment_json
+                manage_msg.execute()
+
+                manage_msg.json = metadata_json
+                manage_msg.execute()
+
+        elif answers['command'] == 'Exit':
+            # Wait until all messages have been delivered
+            sys.stderr.write('%% Waiting for %d deliveries\n' % len(producer))
+            producer.flush()
+            print('That\'s all folks!')
+            run = False
+
+        elif answers['command'] == 'Refresh':
+            clear()
+
+        else:
+            control_json = json.dumps({"category": "control", "run": {"action": answers['command'],
+                                                                      "run_number": run_number,
+                                                                      "date": date}})
+            control_msg = Message(topic_control, control_json)
+            control_msg.execute()
+
+
+def status_readout():
+    global run, daq_state, daq_error, daq_feedback, daq_evt_rate, daq_events, daq_kB_rate
+    while run:
+        msg = consumer.poll(timeout=1.0)
+        if msg is None:
+            continue
+        if msg.error():
+            raise KafkaException(msg.error())
+        else:
+            # Proper message
+            sys.stderr.write('%% %s [%d] at offset %d with key %s:\n' %
+                             (msg.topic(), msg.partition(), msg.offset(), str(msg.key())))
+            json_data = json.loads(msg.value())
+            if json_data['category'] == 'feedback':
+                daq_state = json_data['daq']['run_state']
+                daq_feedback = json_data['os']
+                daq_evt_rate = str(json_data['daq']['evt_rate'])
+                daq_kB_rate = str(json_data['daq']['kB_rate'])
+                daq_events = str(json_data['daq']['events'])
+            elif json_data['category'] == 'errors':
+                daq_error = json_data['msg']
+
+    consumer.close()
+
+
+if __name__ == '__main__':
+    # start in a state to begin execution, and set DAQ state to "normal"
+    run = True

```

Comment:
Again, perhaps put all of this in a main function and just call that from here?

Reply:
If the module is the main program it will run.

Reply:
The main guard in Python is different than in other languages. It determines what gets run under certain circumstances. For example, if you run this by itself them everything in the main guard will execute. E.g. `python3 kafka_pyinq.py`. However if, one were to include this, as in, `import kafka_pyinq` then the code in the main guard will not execute.

Reply:
Oh yes, it's clear that this works as it is. I just question the long-term wisdom of having everything being implicitly global. (I think it's my C background - global variables make me slightly uncomfortable).

Reply:
@caseyc37 : I had the same problem when I started w/ python. It helps to remember that it's a functional scripting language.

Reply:
I would recommend that you find a cleaner way to handle this in the threads. Threads can be monitored with `.isAlive()` and can be controlled via signals or ,if made into a class, controlled via arguments.

Reply:
I've added an exit event signal.